File: back-end/src/main/java/es/ull/project/application/service/workflow/WorkFlowUpdateService.java
package es.ull.project.application.service.workflow;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import org.springframework.beans.factory.annotation.Autowired;

import es.ull.project.application.repository.WorkFlowRepository;
import es.ull.project.application.usecase.workflow.WorkFlowUpdateUseCase;
import es.ull.project.domain.entity.WorkFlow;
import es.ull.project.domain.valueobject.WorkFlowName;
import es.ull.project.domain.valueobject.Node;

public class WorkFlowUpdateService implements WorkFlowUpdateUseCase {

    @Autowired
    private WorkFlowRepository repository;

    @Override
    public Optional<WorkFlow> updateWorkFlow(
            UUID id,
            WorkFlowName name,
            List<Node> nodes,
            boolean enabled) {
        WorkFlow existingFlow = repository.fetchById(id);
        final WorkFlow workflow = new WorkFlow(id, name, nodes, existingFlow.getCreationDate(), enabled);
        return Optional.ofNullable(repository.update(workflow));
    }
}

File: front-end/src/application/service/workflow/workflow-update-service.ts
import type { WorkFlowUpdateUseCase } from '../../usecase/workflow/workflow-update-use-case';
import { WorkFlow } from '../../../domain/entity/workflow';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError, UllUUID } from '@ull-tfg/ull-tfg-typescript';
import type { WorkFlowRepository } from '../../repository/workflow-repository';
import { WorkFlowName } from '@/domain/value-object/workflow-name';
import { Node } from '@/domain/value-object/node';

export class WorkFlowUpdateService implements WorkFlowUpdateUseCase {

    constructor(private readonly flowRepository: WorkFlowRepository) { }

    async updateWorkFlow(
        uuid: UllUUID,
        workFlowName: WorkFlowName,
        nodes: Node[],
        creationDate: Date | string,
        enabled: boolean
    ): Promise<Either<DataError, WorkFlow>> {
        const workflow = new WorkFlow(workFlowName, nodes, enabled, creationDate);
        return this.flowRepository.updateWorkFlow(uuid, workflow);
    }
}
